### ISSUE TITLE
Server stack exec phase prints mangled text

### ISSUE THREAD

**Issue by milesfrain** [purescript/trypurescript#216]
At this stage of the [local dev instructions](https://github.com/purescript/trypurescript/blob/master/README.md#2-local-compile-server-setup):
```
stack exec trypurescript 8081 $(spago sources)
```
I see the following output:
```
e
htw.
aHrTeM
L.CCGooemCmnpopeimirlplaiiitnlnegigd n 
DgHO aMHl.ToHTgTPeMunLr..eSI.vnCgdo.enIxtnerddie
bx.eBdiCCCC
sooooCCcmmmmoooppppmmtiiiipptlllliiiiiiill
nnnniiggggnn    ggSSSH  pppaEEooolllrrrommkkkgii...essHHHnhhttt...mmmHHHlllTTT...MMMPEELLLrll..
oeePSpmmrteeeoyrnnplttteeissrde.
t.sKiG
eeeysne
edCr
oamtpCCCCeioooodlmmmm
ippppniiiigllll iiiiSnnnnpggggo    rHHHHkaaaa.CllllHoooootmggggmpeeeelinnnn
l....iHHHSnTCTTvgMoMMg LmLL.E.p..AlEiPEtmvlrltieioersnnpmihtgeeb.s rnuH
StttTpiseMoe
sLrs
.k.S.AtARypIlpAe

dCCC
ooommmpppiiillCliioinnmnggpg  i HHlHaaiallnloogogg geeSennpn..o.SHrHvTkTgM.M.LPLE
u.lrEeelmACeepomnpmet
pnsit
lsi.nKge yHeadl
ogen
Compiling CHoamlpoigleinn.gA fHfa.lDorgievne.rH
ooks.HookM
CompCiolmipnigl iHnaglCC oooHgmmaepplniio.llgHiieonnnogg.k  HsHHo.aaoIllknoostgg.eeeIrnnnn..taVAelDfr.ofnEm
av.laD.lrU.isTveyeHproe
osk
F
ComCpoimlpiinlgi nHga lHoagleong.eHno.oHkoso.kHso.oIkn
ternal.Eval
Compiling Halogen.Hooks.Component
Compiling Halogen.Hooks
Compiling HalCoCogoCCmCemooponpmmim.ipplpHliiiioillnloniigikgnn ns ggHg.H  a EaHHlHxlaaoatollglrgooeoaeggng.nee.eA.nnHncH..o.toHHoHioookookoosonskk.ks.ssEs.E..x.ExEEtEvtxxrxerttatnarr.rt.aaHasH..o.
oHHoHoookokoososkk.k.ssUsU..s.sUUeUessDsEeeeevGTbSeehotntruat
ont
tceteFlrne


Compiling Halogen.Hooks.Extra.Hooks
Setting phasers to stun... (port 8081) (ctrl-c to quit)
```

**Comment by hdgarrood** [comment]
I suspect this is a bug in the compiler, I don't think we have the capability of fixing this here.

**Comment by JordanMartinez** [comment]
Probably. I remember seeing this in [purescript/purescript#3839 (comment)](https://github.com/purescript/purescript/pull/3839#issuecomment-612117306). Harry mentioned a possible solution, but we tracked it down to [this change](https://github.com/purescript/purescript/pull/3839/files/0fbc58a20ecbe08eaabfa55a7d01e16f82af0efa#r407145758).

**Comment by hdgarrood** [comment]
The solution I suggested is what we have now, and I think that worked at the time, so I guess they must have changed the implementation again.

**Comment by JordanMartinez** [comment]
You also [suggested setting the LineBuffering](https://github.com/purescript/purescript/pull/3839#issuecomment-612532073), which did fix the text mangling. The other change I linked above just fixed the remaining newline issue.

The only place I found `IO.hSetBuffering IO.stdout IO.LineBuffering` in the purescript codebase was in `app/Main.hs`. So, perhaps we need to do the same here in `server/Main.hs`?
